### DISCUSSION TITLE
Is it possible to use background image-set()?

### DISCUSSION THREAD

**Discussion by neoswf** [tailwindlabs/tailwindcss#8271] [Help]
Is it possible to set `image-set()` to be able to display retina background images? 

I have tried to apply the syntax, but without any success:

```
<div class="
bg-[url(
   image-set(
      url('img/icon.png') 1x, 
      url('img/icon@2x.png') 2x
   )
)] 
bg-no-repeat bg-center">Sou parceiro</div>
```

This is still a very experimental feature ([reference:caniuse.com](https://caniuse.com/?search=image-set)).
We need to use a background fallback for browsers which do not offer support and a `-webkit-` prefix for Google Chrome. 

And if it's impossible to implement or use `image-set` right now, what are your recommendations for retina background images using the Tailwind utility-first philosophy, without of-course creating class names outside the HTML scope, where I can apply @media rules:

```
.some-element-with-bg-img{
    background-image: url("img/icon.png");
    background-size: cover;
}

@media (-webkit-min-device-pixel-ratio: 2), 
(min-resolution: 192dpi) { 
    .some-element-with-bg-img {
        background-image: url("img/icon@2x.png");
    }
} 
```

Thank you 🙏

**Comment by niclashoyer** [comment]
@neoswf a bit late to the party, but I just released a proof-of-concept tailwind plugin that adds variants for this: https://github.com/niclashoyer/tailwindcss-resolution

Example:

```html
<div class="bg-[url(img/icon.png)] resolution-2:bg-url(img/icon@2x.png)"></div>
```
